File: javascript/leetcode/minimum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (root === null) { return 0; }
    if (root.left !== null && root.right !== null) {
        return Math.min(minDepth(root.left), minDepth(root.right)) + 1;
    } else {
        return (root.left !== null ? minDepth(root.left) : minDepth(root.right)) + 1;
    }
    
};

File: ruby/leetcode/minimum-depth-of-binary-tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Integer}
def min_depth(root)
    if root == nil
        return 0
    elsif root.left != nil && root.right != nil
        return [min_depth(root.left), min_depth(root.right)].min + 1
    else
        (if root.left then min_depth(root.left) else min_depth(root.right) end) + 1
    end
end
